Fix a broken test:

https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/issues/803

Patches copied from upstream source repository:

https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/commit/2ce5909f3a0b0da3abb7b794215d6b8b72a3b7fa
https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/commit/f5310ce346180a717f091f2f09bcbb3ddfb15436

From 2ce5909f3a0b0da3abb7b794215d6b8b72a3b7fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Thu, 12 Nov 2020 23:38:21 +0000
Subject: [PATCH 1/2] tests: qtdemux: fix crash on 32-bit architectures

Fixes https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/issues/803

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/merge_requests/815>
---
 tests/check/elements/qtdemux.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/tests/check/elements/qtdemux.c b/tests/check/elements/qtdemux.c
index 5271c6576..0c748278b 100644
--- a/tests/check/elements/qtdemux.c
+++ b/tests/check/elements/qtdemux.c
@@ -797,9 +797,10 @@ GST_START_TEST (test_qtdemux_pad_names)
       "protection-system", G_TYPE_STRING,
       "9a04f079-9840-4286-ab92-e65be0885f95", NULL);
   caps =
-      gst_caps_new_simple ("video/quicktime", "variant", G_TYPE_STRING,
-      "mss-fragmented", "timesacle", G_TYPE_UINT64, 10000000, "media-caps",
-      GST_TYPE_CAPS, mediacaps, NULL);
+      gst_caps_new_simple ("video/quicktime",
+      "variant", G_TYPE_STRING, "mss-fragmented",
+      "timesacle", G_TYPE_UINT64, G_GUINT64_CONSTANT (10000000),
+      "media-caps", GST_TYPE_CAPS, mediacaps, NULL);
 
   /* Send segment event* */
   event = gst_event_new_caps (caps);
@@ -852,9 +853,10 @@ GST_START_TEST (test_qtdemux_pad_names)
       "protection-system", G_TYPE_STRING,
       "9a04f079-9840-4286-ab92-e65be0885f95", NULL);
   caps =
-      gst_caps_new_simple ("video/quicktime", "variant", G_TYPE_STRING,
-      "mss-fragmented", "timesacle", G_TYPE_UINT64, 10000000, "media-caps",
-      GST_TYPE_CAPS, mediacaps, NULL);
+      gst_caps_new_simple ("video/quicktime",
+      "variant", G_TYPE_STRING, "mss-fragmented",
+      "timesacle", G_TYPE_UINT64, G_GUINT64_CONSTANT (10000000),
+      "media-caps", GST_TYPE_CAPS, mediacaps, NULL);
 
   /* Send segment event* */
   event = gst_event_new_caps (caps);
-- 
2.30.0


From f5310ce346180a717f091f2f09bcbb3ddfb15436 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Thu, 12 Nov 2020 23:39:21 +0000
Subject: [PATCH 2/2] tests: qtdemux: fix typo in caps field

timesacle -> timescale

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/merge_requests/815>
---
 tests/check/elements/qtdemux.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/check/elements/qtdemux.c b/tests/check/elements/qtdemux.c
index 0c748278b..4a14c45c0 100644
--- a/tests/check/elements/qtdemux.c
+++ b/tests/check/elements/qtdemux.c
@@ -799,7 +799,7 @@ GST_START_TEST (test_qtdemux_pad_names)
   caps =
       gst_caps_new_simple ("video/quicktime",
       "variant", G_TYPE_STRING, "mss-fragmented",
-      "timesacle", G_TYPE_UINT64, G_GUINT64_CONSTANT (10000000),
+      "timescale", G_TYPE_UINT64, G_GUINT64_CONSTANT (10000000),
       "media-caps", GST_TYPE_CAPS, mediacaps, NULL);
 
   /* Send segment event* */
@@ -855,7 +855,7 @@ GST_START_TEST (test_qtdemux_pad_names)
   caps =
       gst_caps_new_simple ("video/quicktime",
       "variant", G_TYPE_STRING, "mss-fragmented",
-      "timesacle", G_TYPE_UINT64, G_GUINT64_CONSTANT (10000000),
+      "timescale", G_TYPE_UINT64, G_GUINT64_CONSTANT (10000000),
       "media-caps", GST_TYPE_CAPS, mediacaps, NULL);
 
   /* Send segment event* */
-- 
2.30.0

